res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -1794,6 +1796,12 @@ TraktorS3.Controller.prototype.deckSwitchHandler = function(field) {
     var channel = this.Channels[field.group];
     var deck = channel.parentDeck;
     deck.activateChannel(channel);
+
+    var isScratching = engine.getValue(deck.previousActiveChannel, "scratch2_enable");
```

Comment:
instead of this new "previouschannel" value, why not disable scratching at the top of the deckswitchhandler, before the active channel gets updated?  so, disable scratching and then activateChannel.

Reply:
hi @ywwg - that's doable and also an easier way of doing it. Will make the change 👍 
thank you